File: output/go/tests/test_mixed_body_strings_comments.go

package main

import "fmt"

type MixedBodyStringsCommentsFrameEvent struct {
    _message    string
    _parameters map[string]any
}

type MixedBodyStringsCommentsFrameContext struct {
    _event  MixedBodyStringsCommentsFrameEvent
    _return any
    _data   map[string]any
}

type MixedBodyStringsCommentsCompartment struct {
    state            string
    stateArgs        map[string]any
    stateVars        map[string]any
    enterArgs        map[string]any
    exitArgs         map[string]any
    forwardEvent     *MixedBodyStringsCommentsFrameEvent
    parentCompartment *MixedBodyStringsCommentsCompartment
}

func newMixedBodyStringsCommentsCompartment(state string) *MixedBodyStringsCommentsCompartment {
    return &MixedBodyStringsCommentsCompartment{
        state:    state,
        stateArgs: make(map[string]any),
        stateVars: make(map[string]any),
        enterArgs: make(map[string]any),
        exitArgs:  make(map[string]any),
    }
}

func (c *MixedBodyStringsCommentsCompartment) copy() *MixedBodyStringsCommentsCompartment {
    nc := &MixedBodyStringsCommentsCompartment{
        state: c.state,
        stateArgs: make(map[string]any),
        stateVars: make(map[string]any),
        enterArgs: make(map[string]any),
        exitArgs:  make(map[string]any),
        forwardEvent:     c.forwardEvent,
        parentCompartment: c.parentCompartment,
    }
    for k, v := range c.stateArgs { nc.stateArgs[k] = v }
    for k, v := range c.stateVars { nc.stateVars[k] = v }
    for k, v := range c.enterArgs { nc.enterArgs[k] = v }
    for k, v := range c.exitArgs { nc.exitArgs[k] = v }
    return nc
}

type MixedBodyStringsComments struct {
    _state_stack []*MixedBodyStringsCommentsCompartment
    __compartment *MixedBodyStringsCommentsCompartment
    __next_compartment *MixedBodyStringsCommentsCompartment
    _context_stack []MixedBodyStringsCommentsFrameContext
}

func NewMixedBodyStringsComments() *MixedBodyStringsComments {
    s := &MixedBodyStringsComments{}
    s._state_stack = make([]*MixedBodyStringsCommentsCompartment, 0)
    s._context_stack = make([]MixedBodyStringsCommentsFrameContext, 0)
    s.__compartment = newMixedBodyStringsCommentsCompartment("Init")
    s.__next_compartment = nil
    __frame_event := MixedBodyStringsCommentsFrameEvent{_message: "$>", _parameters: nil}
    __ctx := MixedBodyStringsCommentsFrameContext{_event: __frame_event, _data: make(map[string]any)}
    s._context_stack = append(s._context_stack, __ctx)
    s.__kernel(&s._context_stack[len(s._context_stack)-1]._event)
    s._context_stack = s._context_stack[:len(s._context_stack)-1]
    return s
}

func (s *MixedBodyStringsComments) __kernel(__e *MixedBodyStringsCommentsFrameEvent) {
    s.__router(__e)
    for s.__next_compartment != nil {
        next_compartment := s.__next_compartment
        s.__next_compartment = nil
        exit_event := &MixedBodyStringsCommentsFrameEvent{_message: "<$", _parameters: s.__compartment.exitArgs}
        s.__router(exit_event)
        s.__compartment = next_compartment
        if s.__compartment.forwardEvent == nil {
            enter_event := &MixedBodyStringsCommentsFrameEvent{_message: "$>", _parameters: s.__compartment.enterArgs}
            s.__router(enter_event)
        } else {
            forward_event := s.__compartment.forwardEvent
            s.__compartment.forwardEvent = nil
            if forward_event._message == "$>" {
                s.__router(forward_event)
            } else {
                enter_event := &MixedBodyStringsCommentsFrameEvent{_message: "$>", _parameters: s.__compartment.enterArgs}
                s.__router(enter_event)
                s.__router(forward_event)
            }
        }
    }
}

func (s *MixedBodyStringsComments) __router(__e *MixedBodyStringsCommentsFrameEvent) {
    switch s.__compartment.state {
    case "Init":
        s._state_Init(__e)
    case "Done":
        s._state_Done(__e)
    }
}

func (s *MixedBodyStringsComments) __transition(next *MixedBodyStringsCommentsCompartment) {
    s.__next_compartment = next
}

func (s *MixedBodyStringsComments) Start() {
    __e := MixedBodyStringsCommentsFrameEvent{_message: "Start"}
    __ctx := MixedBodyStringsCommentsFrameContext{_event: __e, _data: make(map[string]any)}
    s._context_stack = append(s._context_stack, __ctx)
    s.__kernel(&s._context_stack[len(s._context_stack)-1]._event)
    s._context_stack = s._context_stack[:len(s._context_stack)-1]
}

func (s *MixedBodyStringsComments) _state_Init(__e *MixedBodyStringsCommentsFrameEvent) {
    if __e._message == "Start" {
        // Native Go with Frame-statement-like tokens in strings and comments
        text := "This mentions -> $Next and pop$ inside a string."
        // A comment that mentions => $^ and -> $Other should not be parsed as Frame
        fmt.Println(text)
        __compartment := newMixedBodyStringsCommentsCompartment("Done")
        __compartment.parentCompartment = s.__compartment.copy()
        s.__transition(__compartment)
        return
    }
}

func (s *MixedBodyStringsComments) _state_Done(__e *MixedBodyStringsCommentsFrameEvent) {
}

func main() {
	fmt.Println("TAP version 14")
	fmt.Println("1..1")
	_ = NewMixedBodyStringsComments()
	fmt.Println("ok 1 - test_mixed_body_strings_comments")
}

File: output/cpp/tests/test_mixed_body_strings_comments.cpp
#include <string>
#include <unordered_map>
#include <vector>
#include <any>
#include <memory>
#include <functional>


#include <iostream>
#include <string>
#include <cassert>

class MixedBodyStringsCommentsFrameEvent {
public:
    std::string _message;
    std::unordered_map<std::string, std::any> _parameters;

    MixedBodyStringsCommentsFrameEvent(const std::string& message, std::unordered_map<std::string, std::any> params = {})
        : _message(message), _parameters(std::move(params)) {}
};

class MixedBodyStringsCommentsFrameContext {
public:
    MixedBodyStringsCommentsFrameEvent _event;
    std::any _return;
    std::unordered_map<std::string, std::any> _data;

    MixedBodyStringsCommentsFrameContext(MixedBodyStringsCommentsFrameEvent event, std::any default_return = {})
        : _event(std::move(event)), _return(std::move(default_return)) {}
};

class MixedBodyStringsCommentsCompartment {
public:
    std::string state;
    std::unordered_map<std::string, std::any> state_args;
    std::unordered_map<std::string, std::any> state_vars;
    std::unordered_map<std::string, std::any> enter_args;
    std::unordered_map<std::string, std::any> exit_args;
    std::unique_ptr<MixedBodyStringsCommentsFrameEvent> forward_event;
    std::unique_ptr<MixedBodyStringsCommentsCompartment> parent_compartment;

    explicit MixedBodyStringsCommentsCompartment(const std::string& state) : state(state) {}

    std::unique_ptr<MixedBodyStringsCommentsCompartment> clone() const {
        auto c = std::make_unique<MixedBodyStringsCommentsCompartment>(state);
        c->state_args = state_args;
        c->state_vars = state_vars;
        c->enter_args = enter_args;
        c->exit_args = exit_args;
        return c;
    }
};

class MixedBodyStringsComments {
private:
    std::vector<std::unique_ptr<MixedBodyStringsCommentsCompartment>> _state_stack;
    std::unique_ptr<MixedBodyStringsCommentsCompartment> __compartment;
    std::unique_ptr<MixedBodyStringsCommentsCompartment> __next_compartment;
    std::vector<MixedBodyStringsCommentsFrameContext> _context_stack;

    void __kernel(MixedBodyStringsCommentsFrameEvent& __e) {
        __router(__e);
        while (__next_compartment) {
            auto next_compartment = std::move(__next_compartment);
            MixedBodyStringsCommentsFrameEvent exit_event("<$");
            __router(exit_event);
            __compartment = std::move(next_compartment);
            if (!__compartment->forward_event) {
                MixedBodyStringsCommentsFrameEvent enter_event("$>");
                __router(enter_event);
            } else {
                auto forward_event = std::move(__compartment->forward_event);
                if (forward_event->_message == "$>") {
                    __router(*forward_event);
                } else {
                    MixedBodyStringsCommentsFrameEvent enter_event("$>");
                    __router(enter_event);
                    __router(*forward_event);
                }
            }
        }
    }

    void __router(MixedBodyStringsCommentsFrameEvent& __e) {
        const auto& state_name = __compartment->state;
        if (state_name == "Init") {
            _state_Init(__e);
        } else if (state_name == "Done") {
            _state_Done(__e);
        }
    }

    void __transition(std::unique_ptr<MixedBodyStringsCommentsCompartment> next) {
        __next_compartment = std::move(next);
    }

    void _state_Init(MixedBodyStringsCommentsFrameEvent& __e) {
        if (__e._message == "start") {
            // Native C++ with Frame-statement-like tokens in strings and comments
            std::string text = "This mentions -> $Next and pop$ inside a string.";
            // A comment that mentions => $^ and -> $Other should not be parsed as Frame
            printf("%s\n", text.c_str());
            auto __new_compartment = std::make_unique<MixedBodyStringsCommentsCompartment>("Done");
            __new_compartment->parent_compartment = __compartment->clone();
            __transition(std::move(__new_compartment));
            return;
        }
    }

    void _state_Done(MixedBodyStringsCommentsFrameEvent& __e) {

    }

public:
    MixedBodyStringsComments() {
        __compartment = std::make_unique<MixedBodyStringsCommentsCompartment>("Init");
        MixedBodyStringsCommentsFrameEvent __frame_event("$>");
        MixedBodyStringsCommentsFrameContext __ctx(std::move(__frame_event));
        _context_stack.push_back(std::move(__ctx));
        __kernel(_context_stack.back()._event);
        _context_stack.pop_back();
    }

    void start() {
        MixedBodyStringsCommentsFrameEvent __e("start");
        MixedBodyStringsCommentsFrameContext __ctx(std::move(__e));
        _context_stack.push_back(std::move(__ctx));
        __kernel(_context_stack.back()._event);
        _context_stack.pop_back();
    }
};

// Stub functions for placeholder calls
void native() {}
void x() {}

// TAP test harness
int main() {
    printf("TAP version 14\n");
    printf("1..1\n");
    try {
        MixedBodyStringsComments s;
        s.start();
        printf("ok 1 - test_mixed_body_strings_comments\n");
    } catch (...) {
        printf("not ok 1 - test_mixed_body_strings_comments\n");
    }
    return 0;
}
